docs/proposals/config-dump.md

```diff
@@ -0,0 +1,248 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should refer to the docs that go with that version.
+
+Documentation for other releases can be found at [releases.K8s.io](http://releases.K8s.io).
+</strong>
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+
+# Overview
+
+Here we describe a proposal for retrieving configurations (configs) that govern the behavior of any one of the Kubernetes (K8s) components. By components we mean any of the following: apiserver, scheduler, control-manager, and kubelet. We expect that these configurations be for the most part static through the life of the component but fully understand that changes from specs may affect these. Consequently, any retrieval of the configurations is to be considered true and valid at the time of retrieval and we will assume that the user is well aware of any transient behavior.   
+
+We propose several important changes:
+* new kubectl command: 
+    * `kubectl get componentconfigs <componentName> (alias compcs) --id <componentId>`
```

Comment:
This seems like the wrong way to do this.

We already have configmaps in the system. System components should be using configmaps rather than flags for their configuration.

That handles where this state ought to be stored. The next part of the problem is enumerating the system components. System components ought to (IMO) be running in pods registered with the system. So this should be as easy as listing pods in the kube-system namespace.

The next part of the problem is figuring out which configmap(s) each system pod is getting its config from. You can do this just by looking at the system pod.

Basically, this problem is already solved by existing Kubernetes primitives, and we should just use these. If the primitives don't work for this case, we should fix them. If it's hard to use, we should make it easier. If we want to build a "get configuration" kubectl command, it should do this by looking up a configmap that a pod is using--it should work for all pods, not just system components.

IMO we should *not* add a special configuration mechanism for kubernetes system components.